content/en/docs/How-to guides/kaniko-build-push.md

```diff
@@ -0,0 +1,411 @@
+<!--
+---
+title: "Build and push an image with Tekton"
+linkTitle: "Build and push an image with Tekton"
+weight: 2
+description: >
+  Create a Pipeline to fetch the source code, build, and push an image with
+  Kaniko and Tekton.
+---
+-->
+
+This guide shows you how to:
+
+1.  Create a Task to clone source code from a git repository.
+1.  Create a second Task to use the cloned code to build a Docker image and push
+    it to a registry.
+
+If you are already familiar with Tekton and just want to see the example, you
+can [skip to the full code samples](#full-code-samples).
+
+## Prerequisites
+
+1.  To follow this How-to you must have a Kubernetes cluster up and running and
+    [kubectl][kubectl] properly configured to issue commands to your cluster.
+
+1.  Install Tekton Pipelines:
+    
+    ```bash
+    kubectl apply --filename \
+    https://storage.googleapis.com/tekton-releases/pipeline/latest/release.yaml
+    ```
+
+    See the [Pipelines installation documentation][pipelines-inst] for other
+    installation options.
+
+1.  Install the [Tekton CLI, `tkn`][tkn-inst], on your machine.
+
+If this is your first time using Tekton Pipelines, we recommend that you
+complete the [Getting Started tutorials][getting-started] before proceeding with
+this guide.
+
+[pipelines-inst]: /docs/pipelines/install/
+[tkn-inst]: /docs/cli/
+[getting-started]: /docs/getting-started/
+[kubectl]: https://kubernetes.io/docs/tasks/tools/#kubectl
+
+## Clone the repository
+
+Create a new Pipeline, `pipeline,yaml`, that uses the *git clone* Task to [clone
+the source code from a git repository][tekton-clone]:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: clone-build-push
+spec:
+  description: | 
+    This pipeline clones a git repo, builds a Docker image with Kaniko and
+    pushes it to a registry
+  params:
+  - name: repo-url
+    type: string
+  workspaces:
+  - name: shared-data
+  tasks:
+  - name: fetch-source
+    taskRef:
+      name: git-clone
```

Comment:
For this to work, the git-clone task has to be installed on the cluster from the catalog right? Should we add a step that mentions that?

Reply:
Ahh I see, the link is to the other doc on cloning a repo

Reply:
Turns out, this already exists, just in the section on running the pipeline